src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java

```diff
@@ -106,17 +114,105 @@ public final long pendingSidelineMessagesCount() {
         return Long.MAX_VALUE;
     }
 
+    /**
+     * Note: Timeout is in MilliSeconds, Function take a list of message as input and return not ack msg as output
+     * @param messages : Messages to be published
+     * @param properties
+     * @param timeout : in MS timeout for waiting on countDownLatch
+     * @param unit : timeout unit
+     * @return : List of message nacked
+     * @throws Exception
+     */
+    public List<Message> publishWithConfirmListener(List<Message> messages, AMQP.BasicProperties properties,
+            long timeout, @NotNull TimeUnit unit) throws Exception {
+        publishChannel.confirmSelect();
+        ConcurrentNavigableMap<Long, Message> outstandingConfirms = new ConcurrentSkipListMap<>();
+        List<Message> nackedMessages = new ArrayList<>();
+        CountDownLatch publishAckLatch = new CountDownLatch(messages.size());
+
+        publishChannel.addConfirmListener((sequenceNumber, multiple) -> {
+            messagesAck(sequenceNumber, multiple, outstandingConfirms, publishAckLatch);
+        }, (sequenceNumber, multiple) -> {
+            nackedMessages.addAll(messagesNack(sequenceNumber, multiple, outstandingConfirms, publishAckLatch));
```

Comment:
Why not just assign the return value rather than this?

Reply:
Because there can be multiple callback with multiple = true

Reply:
Ah, my bad